cardano-cli/test/cardano-cli-test/Test/Cli/Hash.hs

```diff
@@ -93,3 +107,63 @@ hprop_generate_anchor_data_hash_from_file_uri =
       [] -> do
         H.note_ ("Path doesn't split:" ++ path)
         H.failure
+
+-- | Execute me with:
+-- @cabal test cardano-cli-test --test-options '-p "/check anchor data hash from http uri/"'@
+hprop_check_anchor_data_hash_from_http_uri :: Property
+hprop_check_anchor_data_hash_from_http_uri =
+  propertyOnce $ do
+    let relativeUrl = ["example", "url", "file.txt"]
+    serveFileWhile
+      relativeUrl
+      exampleAnchorDataPath
+      ( \port ->
+          void $
+            execCardanoCLI
+              [ "hash"
+              , "anchor-data"
+              , "--url"
+              , "http://localhost:" ++ show port ++ "/" ++ intercalate "/" relativeUrl
+              , "--expected-hash"
+              , exampleAnchorDataHash
+              ]
+      )
+
+-- | Takes a relative url (as a list of segments), a file path, and an action, and it serves
+-- the file in the url provided in a random free port that is passed as a parameter to the
+-- action. After the action returns, it shuts down the server. It returns the result of the
+-- action. It also ensures the server is shut down even if the action throws an exception.
+serveFileWhile
+  :: (MonadBaseControl IO m, MonadTest m, MonadIO m, HasCallStack)
+  => [String]
+  -- ^ Relative URL where the file will be served.
+  -- Each element is a segment of the URL.
+  -> FilePath
+  -- ^ File path for the file to serve
+  -> (Int -> m a)
+  -- ^ Action to run while the file is being served.
+  -- It receives the port the server is listening on
+  -> m a
+serveFileWhile relativeUrl filePath action =
+  bracket
+    -- Server setup (resource acquisition)
+    ( do
+        -- Get the port the server is listening on
+        (port, socket) <- H.evalIO openFreePort
+        -- Serve the file
+        let app :: Request -> (Response -> IO ResponseReceived) -> IO ResponseReceived
+            app req respond = do
+              let path = T.unpack <$> pathInfo req
+              if path == relativeUrl
+                -- Status -> ResponseHeaders -> FilePath -> Maybe FilePart -> Response
+                then respond $ responseFile status200 [("Content-Type", "text/plain")] filePath Nothing
+                else respond $ responseLBS status404 [("Content-Type", "text/plain")] "404 - Not Found"
```

Comment:
Rendering the not found url would be useful

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/895/commits/198dcbce37e5a3db3e66311d025a9d6b452ebf74